src/org/parosproxy/paros/core/scanner/VariantHeader.java

```diff
@@ -33,14 +35,19 @@
  */
 public class VariantHeader implements Variant {
 
-    // I've found an XSS using this payload on the Host header
-    // "%s-->\">'>'\"<sfi%06uv%06u>"
-
-    public static final String[] injectableHeaders = {
-        HttpRequestHeader.USER_AGENT,
-        HttpRequestHeader.REFERER,
-        HttpRequestHeader.HOST
+    public static final String[] nonInjectableHeaders = {
+        HttpRequestHeader.CONTENT_LENGTH,	//scanning this would likely break the entire request 
+        HttpRequestHeader.PRAGMA,			//unlikely to be picked up/used by the app itself.
+        HttpRequestHeader.CACHE_CONTROL,	//unlikely to be picked up/used by the app itself.          
+        HttpRequestHeader.COOKIE,			//The Cookie header has its own variant that controls whether it is scanned. Better not to scan it as a header.
+        HttpRequestHeader.AUTHORIZATION,	//scanning this would break authorisation
+        "PROXY-AUTHORIZATION",				//scanning this would break authorisation		//TODO: add to HttpRequestHeader?

```

Comment:
Yes, better add now to one of the header classes (same for following headers). I'm OK with `HttpRequestHeader`.